src/decorations/intervals.jl

```diff
@@ -29,44 +29,51 @@ of a previously executed sequence of functions acting on an initial interval.
 struct DecoratedInterval{T}
     interval::Interval{T}
     decoration::DECORATION
+
+    function DecoratedInterval{T}(I::Interval, d::DECORATION) where T
+        dd = min(d, decoration(I))
+        dd == ill && return new{T}(Interval(T(NaN), T(NaN)), ill)
+        return new{T}(I, dd)
+    end
 end
 
+DecoratedInterval(I::Interval{T}, d::DECORATION) where T = DecoratedInterval{T}(I, d)
+
 DecoratedInterval(I::DecoratedInterval, dec::DECORATION) = DecoratedInterval(I.interval, dec)
 
-function DecoratedInterval(a::Real, b::Real, d::DECORATION)
-    is_valid_interval(a, b) || return DecoratedInterval(Interval(a,b), ill)
-    return DecoratedInterval(Interval(a,b), d)
-end
+DecoratedInterval(a::Real, b::Real, d::DECORATION) = DecoratedInterval(Interval(a, b), d)
 
 DecoratedInterval(a::Real, d::DECORATION) = DecoratedInterval(a, a, d)
 DecoratedInterval(a::Tuple, d::DECORATION) = DecoratedInterval(a..., d)
 
 function DecoratedInterval{T}(I::Interval) where {T}
     d = decoration(I)
-    d <= trv && return DecoratedInterval{T}(I, d)
-    d == ill && return DecoratedInterval{T}(nai(I), d)
     return DecoratedInterval{T}(I, d)
 end
 
-DecoratedInterval(I::Interval) = DecoratedInterval{default_bound()}(I)
+DecoratedInterval(I::Interval{T}) where T = DecoratedInterval{T}(I)
 
-function DecoratedInterval(a::Real, b::Real)
-    is_valid_interval(a, b) || return DecoratedInterval(Interval(a,b), ill)
-    return DecoratedInterval(Interval(a,b))
-end
+DecoratedInterval(a::Real, b::Real) = DecoratedInterval(Interval(a,b))
 
 DecoratedInterval(a::Real) = DecoratedInterval(a, a)
 DecoratedInterval(a::Tuple) = DecoratedInterval(a...)
 
-interval(x::DecoratedInterval) = x.interval
+function interval(x::DecoratedInterval{T}) where {T}
+    if isnai(x)
+        @warn "trying to access interval part of [NaI], returning empty interval"
```

Comment:
```suggestion
        @warn "trying to access interval part of [NaI], returning empty interval. Use `x.interval` for unsafe and unchecked access."
```

Reply:
Maybe start the sentence with a capital letter... (Sorry for being too picky.)